client/state/checklist/reducer.js

```diff
@@ -1,28 +1,58 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * Internal dependencies
  */
 import { combineReducers, keyedReducer } from 'state/utils';
 import {
 	JETPACK_MODULE_ACTIVATE_SUCCESS,
+	JETPACK_MODULE_DEACTIVATE_SUCCESS,
 	SITE_CHECKLIST_RECEIVE,
 	SITE_CHECKLIST_TASK_UPDATE,
 } from 'state/action-types';
 import { items as itemSchemas } from './schema';
 
-const markChecklistTaskComplete = ( state, taskId ) => ( {
+const toggleChecklistTask = ( state, taskId ) => ( {
 	...state,
-	tasks: { ...state.tasks, [ taskId ]: true },
+	tasks: {
+		...state.tasks,
+		[ taskId ]: { completed: ! get( state.tasks, [ taskId, 'completed' ], false ) },
+	},
 } );
 
+const moduleTaskMap = {
+	lazy_images: 'jetpack_lazy_images',
+	monitor: 'jetpack_monitor',
+	// Both photon and photon-cdn mark the Site Accelerator Task as completed
+	photon: 'jetpack_site_accelerator',
+	'photon-cdn': 'jetpack_site_accelerator',
+	search: 'jetpack_search',
+	videopress: 'jetpack_video_hosting',
+};
+
 function items( state = {}, action ) {
 	switch ( action.type ) {
 		case SITE_CHECKLIST_RECEIVE:
 			return action.checklist;
 		case SITE_CHECKLIST_TASK_UPDATE:
-			return markChecklistTaskComplete( state, action.taskId );
+			return toggleChecklistTask( state, action.taskId );
 		case JETPACK_MODULE_ACTIVATE_SUCCESS:
-			if ( action.moduleSlug === 'monitor' ) {
-				return markChecklistTaskComplete( state, 'jetpack_monitor' );
+			if ( moduleTaskMap.hasOwnProperty( action.moduleSlug ) ) {
+				return toggleChecklistTask( state, moduleTaskMap[ action.moduleSlug ] );
+			}
+			break;
+		case JETPACK_MODULE_DEACTIVATE_SUCCESS:
+			if ( action.moduleSlug === 'photon' || action.moduleSlug === 'photon-cdn' ) {
+				// We can't know if the other module is still active, so we don't change
+				// Site Accelerator task completion state.
```

Comment:
If we'd decide that it's enough to enable one of these features to complete the task, we could return here just fine?

Reply:
We're actually doing that [here](https://github.com/Automattic/wp-calypso/pull/33710/files#diff-0c8c68f71be23c8ded46a59c076439c1R44) -- but this is about _de_activating, which unfortunately isn't as easy.